File: src/main/java/com/hazza/LeetCodeByJava/No88MergeSortedArray.java
package com.hazza.LeetCodeByJava;

/**
 * Created with IntelliJ IDEA.
 * Description: https://leetcode.com/problems/merge-sorted-array/description/
 * Author: HazzaCheng
 * Contact: hazzacheng@gmail.com
 * Date: 2017-09-25
 * Time: 9:02 AM
 */
public class No88MergeSortedArray {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int count = m + n, i = m - 1, j = n - 1;
        while (count > 0)
            if (j < 0 || (i >= 0 && nums1[i] >= nums2[j])) nums1[--count] = nums1[i--];
            else nums1[--count] = nums2[j--];
    }
}

File: src/main/python/leetcode_by_python/two_pointers/No88_Merge_Sorted_Array.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# Created by HazzaCheng on 2019-07-26
# https://leetcode.com/problems/merge-sorted-array/
from typing import List


class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        i, j, cur = m - 1, n - 1, m + n - 1

        while cur >= 0:
            if j < 0 or (i >= 0 and nums1[i] > nums2[j]):
                nums1[cur] = nums1[i]
                i -= 1
            else:
                nums1[cur] = nums2[j]
                j -= 1
            cur -= 1
